pkg/app/controller.go

```diff
@@ -153,15 +169,47 @@ func (f *HostForwarder) Handler() AuthHTTPHandler {
 		if host == "" {
 			return fmt.Errorf("invalid url missing host value: %q", r.URL.String())
 		}
+		resource := vars["resource"]
+		if resource == "" {
+			return fmt.Errorf("invalid url missing host resource: %q", r.URL.String())
+		}
 		proxy, err := f.buildReverseProxy(zone, host)
 		if err != nil {
 			return err
 		}
+		if f.shouldInjectCredentials[r.Method+":"+resource] {
+			if err := f.InjectCredentials(r, user); err != nil {
+				return err
+			}
+		}
 		proxy.ServeHTTP(w, r)
 		return nil
 	}
 }
 
+func (f *HostForwarder) InjectCredentials(r *http.Request, user UserInfo) error {
+	credentials, err := f.getCredentials(user)
+	if err != nil {
+		return err
+	}
+	if credentials == "" {
+		// TODO(jemoreira) return unauthorized instead when no credentials are found
+		return nil
+	}
+	msg := make(map[string]interface{})
+	if err := json.NewDecoder(r.Body).Decode(&msg); err != nil {
+		return err
+	}
+	msg["Credentials"] = credentials
+	buf, err := json.Marshal(msg)
+	if err != nil {
+		return err
+	}
+	r.ContentLength = int64(len(buf))
```

Comment:
An alternative is injecting the build api credentials as a request header, is this a viable solution? Modifying the body expects the body to be JSON.

Reply:
I thought about that, but I only need to inject credentials for android ci builds, not user builds. Differentiating between those requires parsing the body, so might as well just modify it too.

Reply:
we are already parsing the body, in this case I'd suggest to import `CreateCVDRequest` https://github.com/google/android-cuttlefish/blob/main/frontend/src/liboperator/api/v1/messages.go#L64 and work with the type to avoid errors modifying the payload.

Do `msg["Credentials"] = credentials` is referencing this field https://github.com/google/android-cuttlefish/blob/main/frontend/src/liboperator/api/v1/messages.go#L90? if so, I think the field path would be different.

Reply:
I understand we want to keep this proxy logic as minimal, but in this case I think it's fine to be more robust and transparent about modifying specific request payloads

Reply:
yeah, this pr didn't work when I tested it e2e. I'm working on these fixes atm.

Reply:
done